### PR TITLE
feat: integrate new core logic to Cocos game

### PR DESCRIPTION
**Pull request by tanlethanh**

### CODE REVIEW

File: game/assets/scripts/core/handlers/index.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+			.exec();
+		if (cardsInGround.length === ecs.config.groundSize) {
+			throw GroundLimitReachedError();
+		}
+
+		const groundIndexes = cardsInGround.map(
+			(c) => c.getComponent(ComponentType.CardPlace).index,
+		);
+		const max = Math.max(...groundIndexes);
+		const min = Math.max(...groundIndexes);
```
[the remark this answers is not in the document]
Comment (tanlethanh) [reply game/assets/scripts/core/handlers/index.ts]: yep my fault, resolved

File: game/assets/scripts/core/handlers/index.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	UnexpectedTurnActionError,
+} from '../error';
+import type { EndTurnEvent, SummonCardEvent } from '../events';
+
+export const endTurnEventHandler = () => {
+	const handle = (ecs: ECS, event: EndTurnEvent) => {
+		const { type, playerEid } = event;
+
+		const player = ecs.queryById(event.playerEid);
+		if (player) {
```
Comment (FME849) [inline game/assets/scripts/core/handlers/index.ts:21]: I think it has to be `if (!player)` here.

File: game/assets/scripts/core/handlers/index.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+			throw PlayerEntityNotFoundError(playerEid);
+		}
+
+		const playerId = player.getComponent(ComponentType.PlayerAttribute).id;
+		if (ecs.state.turnOf !== playerId) {
+			throw UnexpectedTurnActionError(type, playerId, ecs.state.turnOf);
+		}
+
+		ecs.state.phase = DuelPhase.PreFight;
```
Comment (FME849) [inline game/assets/scripts/core/handlers/index.ts:30]: we will change phase if `ecs.state.turnof` is second player id, if it's first player id, we have to change turn, not phase

Comment (tanlethanh) [reply game/assets/scripts/core/handlers/index.ts:30]: I think we should have a system to change turn after going over all other systems

**Review by FME849 — APPROVED** [review]

**Pull request merged**